include/nuttx/wdt_notifier.h

```diff
@@ -0,0 +1,97 @@
+/****************************************************************************
+ * include/nuttx/wdt_notifier.h
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_WDT_NOTIFIER_H
+#define __INCLUDE_NUTTX_WDT_NOTIFIER_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/notifier.h>
+#include <nuttx/sched.h>
+
+#include <sys/types.h>
+
+#ifdef CONFIG_WATCHDOG_TIMEOUT_NOTIFIER
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+enum wdt_type_e
```

Comment:
change to macro and remove enum

Reply:
I haved changed this enum to macro and move them into `timers/watchdog.h`
```C
#define WATCHDOG_KEEPALIVE_BY_ONESHOT      0
#define WATCHDOG_KEEPALIVE_BY_TIMER        1
#define WATCHDOG_KEEPALIVE_BY_WDOG         2
#define WATCHDOG_KEEPALIVE_BY_WORKER       3
#define WATCHDOG_KEEPALIVE_BY_CAPTURE      4
#define WATCHDOG_KEEPALIVE_BY_IDLE         5
```